packages/builder-util/src/util.ts

```diff
@@ -121,6 +121,10 @@ export function exec(file: string, args?: Array<string> | null, options?: ExecFi
           }
           resolve(stdout.toString())
         } else {
+          // https://github.com/npm/npm/issues/17624
+          if ((file === "npm" || file === "npm.cmd") && args?.includes("--silent")) {
+            resolve(stdout.toString())
```

Comment:
Just to confirm, this is intentional? We're still resolving with the stdout in silent mode?

Reply:
Yes, the `npm list` command can throw errors in some cases, but these errors don't affect the result. Therefore, we still need to return the stdout, which contains the dependency tree and its paths. 

However, the way it was written before could be a bit misleading, so I've made some adjustments.